config/routes.rb

```diff
@@ -1,31 +1,82 @@
 Triannon::Engine.routes.draw do
 
-  resources :annotations, except: [:update, :edit],
-    # show action must explicitly forbid "new", "iiif" and "oa" as id values;  couldn't
-    #   figure out how to do it with regexp constraint since beginning and end regex
-    #   matchers aren't allowed when enforcing formats for segment (e.g. :id)
-    constraints: lambda { |request|
-                            id = request.env["action_dispatch.request.path_parameters"][:id]
-                            id !~ /^iiif$/ && id !~ /^oa$/ && id !~ /^search$/
-                 } do
-    collection do
-      get 'search', to: 'search#find'
-    end
-  end
+  # 1. can't use resourceful routing because of :anno_root (dynamic path segment)
 
+  # 2. couldn't figure out how to exclude specific values with regexp constraint since beginning and end regex matchers
+  #    aren't allowed when enforcing formats for path segment (i.e. :anno_root, :id)
+
+  # get -> new action
+  get '/annotations/:anno_root/new', to: 'annotations#new'
+  get '/:anno_root/new', to: 'annotations#new',
+    constraints: lambda { |request|
+      anno_root = request.path_parameters[:anno_root]
+      id = request.path_parameters[:id]
+      anno_root !~ /^annotations$/ && anno_root !~ /^search$/ && anno_root !~ /^new$/ && id !~ /^search$/ && id !~ /^new$/
+    }
+
+  # get -> search controller find action
+  get '/annotations/:anno_root/search', to: 'search#find'
+  get '/annotations/search', to: 'search#find'
+  get '/:anno_root/search', to: 'search#find',
+    constraints: lambda { |request|
```

Comment:
Could/should any of these `:constraints` use the same lambda? I think there's a routing syntax where you can wrap multiple routes with some defaults, or you could extract the lambda out into a method/constant somewhere